lib/connect/index.js

```diff
@@ -62,14 +62,20 @@ function connect (brokerUrl, opts) {
       throw new Error('Missing protocol')
     }
     var parsed = new URL(brokerUrl)
+
     // the URL object is a bit special, so copy individual
     // items to the opts object
-    opts.hostname = parsed.hostname
+    opts.hash = parsed.hash
     opts.host = parsed.host
-    opts.protocol = parsed.protocol
+    opts.hostname = parsed.hostname
+    opts.href = parsed.href
+    opts.origin = parsed.origin
+    opts.pathname = parsed.pathname
```

Comment:
Hi 👋 

I just upgraded the version of MQTT.js I'm using to the latest one (4.2.3).
Now I have an error while connecting because the path is not being included in the URL, even after this PR.
Did a few tests and added an extra `opts.path = parsed.pathname` and now the path is passed down and it connects. Can you take a look?
For example, an URL like: `wss://example.server/mqtt` would try to connect to `wss://example.server`.
Thank you!